src/providers/token-lists.provider.ts

```diff
@@ -21,121 +12,101 @@ export interface TokenListsState {
-      }
-
-      lsSet(localStorageKeys.TokenLists.Toggled, state.activeListKeys);
-    }
+/**
+ * STATE
+ */
+const state: TokenListsState = reactive({
+  activeListKeys: [uris.Balancer.Default],
+});
+
+let allTokenLists = {};
+try {
+  allTokenLists = require<TokenListMap>(`/public/data/tokenlists/tokens-${networkId.value}.json`);
+} catch (error) {
+  console.error('Failed to fetch tokenlists', error);
+  throw error;
+}
 
-    /**
-     * Given a token list URI checks if the related token
-     * list has been toggled via the token search modal.
-     */
-    function isActiveList(uri: string): boolean {
-      return state.activeListKeys.includes(uri);
-    }
+/**
+ * All active (toggled) tokenlists
+ */
+const activeTokenLists = computed(
+  (): TokenListMap => pick(allTokenLists, state.activeListKeys)
+);
+
+/**
+ * The default Balancer token list.
+ */
+const defaultTokenList = computed(
+  (): TokenList => allTokenLists[uris.Balancer.Default]
+);
+
+/**
+ * The Balancer vetted token list, contains LBP tokens.
+ */
+const vettedTokenList = computed(
+  (): TokenList => allTokenLists[uris.Balancer.Vetted]
+);
+
+/**
+ * All Balancer token lists mapped by URI.
+ */
+const balancerTokenLists = computed(
+  (): TokenListMap => pick(allTokenLists, uris.Balancer.All)
+);
+
+/**
+ * Approved token lists mapped by URI.
+ * Approved means tokens are compliant and can be presented in the UI.
+ * This excludes lists like the Balancer vetted list.
+ */
+const approvedTokenLists = computed(
+  (): TokenListMap => pick(allTokenLists, uris.Approved)
+);
+
+/**
+ * Adds a token list to the active lists which
+ * makes additonal tokens available in the token search modal.
+ */
+function toggleTokenList(uri: string): void {
+  if (!uris.Approved.includes(uri)) return;
+
+  if (state.activeListKeys.includes(uri)) {
+    // Deactivate token list
+    state.activeListKeys.splice(state.activeListKeys.indexOf(uri), 1);
+  } else {
+    // Activate token list
+    state.activeListKeys.push(uri);
+  }
+
+  lsSet(localStorageKeys.TokenLists.Toggled, state.activeListKeys);
+}
 
-    provide(TokenListsProviderSymbol, {
-      // state
-      ...toRefs(state),
-      // computed
-      allTokenLists,
-      activeTokenLists,
-      defaultTokenList,
-      balancerTokenLists,
-      approvedTokenLists,
-      vettedTokenList,
-      // methods
-      toggleTokenList,
-      isActiveList,
-    });
+/**
+ * Given a token list URI checks if the related token
+ * list has been toggled via the token search modal.
+ */
+function isActiveList(uri: string): boolean {
+  return state.activeListKeys.includes(uri);
+}
 
-    return () => slots.default();
-  },
-};
+export function getTokensListProvision() {
+  return {
+    // state
+    ...toRefs(state),
+    // computed
+    allTokenLists,
+    activeTokenLists,
+    defaultTokenList,
+    balancerTokenLists,
+    approvedTokenLists,
+    vettedTokenList,
+    // methods
+    toggleTokenList,
+    isActiveList,
+  };
+}
```

Comment:
Why does this need to be a function? Also the Provision part of `getTokensListProvision` doesn't really make sense. What about:
```suggestion
export const tokenListsProvider = {
    // state
    ...toRefs(state),
    // computed
    allTokenLists,
    activeTokenLists,
    defaultTokenList,
    balancerTokenLists,
    approvedTokenLists,
    vettedTokenList,
    // methods
    toggleTokenList,
    isActiveList,
}
```

Reply:
Yeah, I kept the two original files (`foo.provider.ts` and `useFoo.ts`) to make the code review simpler but the idea would be deleting `foo.provider.ts` with a const as you suggest.